### PR TITLE
Tax calculation formula change on invoice print and sales detail view

### PR DESCRIPTION
**Pull request by FoishaalAhmed**

### CODE REVIEW

File: resources/views/sale_pos/partials/sale_line_details.blade.php [hunk 1]
```diff
@@ -100,10 +109,17 @@
                 @if($sell_line->line_discount_type == 'percentage') ({{$sell_line->line_discount_amount}}%) @endif
             </td>
             <td>
+                @php
+                    $calTax = $sell_line->product->tax == 1 ? $subTotal - ($subTotal /1.16) : 0;
```
Comment (Nobeul) [inline resources/views/sale_pos/partials/sale_line_details.blade.php]: why $sell_line->product->tax is compared with static value(1)? That is not understandable.

File: resources/views/sale_pos/receipts/slim2.blade.php [hunk 2]
```diff
@@ -445,7 +455,11 @@
                 	<strong>{!! $receipt_details->subtotal_label !!}</strong>
                 </p>
                 <p class="width-50 text-right">
-                	<strong>KSh {{(float)preg_replace('/[^0-9.]/', '', $receipt_details->subtotal) - (float)preg_replace('/[^0-9.]/', '', $receipt_details->taxes[array_key_last($receipt_details->taxes)])}}</strong>
+                	{{-- <strong>KSh {{(float)preg_replace('/[^0-9.]/', '', $receipt_details->subtotal) - (float)preg_replace('/[^0-9.]/', '', $receipt_details->taxes[array_key_last($receipt_details->taxes)])}}</strong> --}}
+
+                	<strong>KSh {{number_format((float)preg_replace('/[^0-9.]/', '', $receipt_details->total) - (float)$totalTax, 2)}}</strong>
```
Comment (Nobeul) [inline resources/views/sale_pos/receipts/slim2.blade.php]: please create a helper function for preg_replace() and use that function calling with arguments.

Comment (FoishaalAhmed) [reply resources/views/sale_pos/receipts/slim2.blade.php:460]: $sell_line->product->tax == 1 is to determine wheather this product accept tax or it is zero tax product. 1= tax applicable, 2= zero tax

**Comment by FoishaalAhmed** [comment]
this is how preg_replace() is used in the whole invoice system, refactoring one will not help. If want to refactor preg_replace(), then need to refactor the whole invoice system

File: resources/views/sale_pos/partials/sale_line_details.blade.php (hunk 1 above)
Comment (FoishaalAhmed) [reply resources/views/sale_pos/partials/sale_line_details.blade.php]: $sell_line->product->tax == 1 is to determine wheather this product accept tax or it is zero tax product. 1= tax applicable, 2= zero tax

File: resources/views/sale_pos/receipts/slim2.blade.php (hunk 2 above)
Comment (FoishaalAhmed) [reply resources/views/sale_pos/receipts/slim2.blade.php]: this is how preg_replace() is used in the whole invoice system, refactoring one will not help. If want to refactor preg_replace(), then need to refactor the whole invoice system

File: resources/views/sale_pos/partials/sale_line_details.blade.php (hunk 1 above)
Comment (Nobeul) [reply resources/views/sale_pos/partials/sale_line_details.blade.php]: Then use constants.

File: resources/views/sale_pos/receipts/slim2.blade.php (hunk 2 above)
Comment (Nobeul) [reply resources/views/sale_pos/receipts/slim2.blade.php]: do the needful please. it won't take a long time. need to use the help of ide.

Comment (FoishaalAhmed) [reply resources/views/sale_pos/receipts/slim2.blade.php]: done. check now

File: resources/views/sale_pos/partials/sale_line_details.blade.php (hunk 1 above)
Comment (FoishaalAhmed) [reply resources/views/sale_pos/partials/sale_line_details.blade.php]: constant added

**Review by Nobeul — APPROVED** [review]

**Pull request merged**